core/src/main/java/hudson/model/ChoiceParameterDefinition.java

```diff
@@ -104,6 +161,19 @@ public String getHelpFile() {
             return "/help/parameter/choice.html";
         }
 
+        @Override
+        /*
+         * We need this for JENKINS-26143 -- reflective creation cannot handle setChoices(Object). See that method for context.
+         */
+        public ParameterDefinition newInstance(@Nullable StaplerRequest req, @Nonnull JSONObject formData) throws FormException {
+            String name = formData.getString("name");
+            String desc = formData.getString("description");
+            String choiceText = formData.getString("choices");
+            ChoiceParameterDefinition parameterDefinition = new ChoiceParameterDefinition(name, desc);
+            parameterDefinition.setChoicesText(choiceText);
```

Comment:
Why not just use the original constructor?

Reply:
+1